### PR TITLE
Modernize Python 2 code to get ready for Python 3

### PR DESCRIPTION
**Pull request by cclauss**
*Issue #, if available:*

*Description of changes:*
* print() is a function in Python 3
* Replace instances of __basestring__ with __six.string_types__

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/apps/lex-make-appointment/lambda_function.py [hunk 1]
```diff
@@ -191,7 +192,7 @@ def lambda_handler(event, context):
                   args['arn'] + "- checking event details...")
             # Identify ARN in event detail
             for key, value in event['detail'].iteritems():
-                if isinstance(value, basestring):
+                if isinstance(value, string_types):
```
Comment (sanathkr) [inline examples/apps/lex-make-appointment/lambda_function.py]: Just a Python clarification question: `string_types` will cover both string and unicode in Python 2 right?

File: samtranslator/model/eventsources/push.py [hunk 2]
```diff
@@ -129,7 +130,7 @@ def to_cloudformation(self, **kwargs):
         :rtype: list
         """
         function = kwargs.get('function')
-        
+
```
Comment (sanathkr) [inline samtranslator/model/eventsources/push.py]: sweeeeeeet!

File: examples/apps/lex-make-appointment/lambda_function.py (hunk 1 above)
Comment (cclauss) [reply examples/apps/lex-make-appointment/lambda_function.py]: Correct.  http://six.readthedocs.io/#six.string_types

File: samtranslator/model/eventsources/push.py (hunk 2 above)
Comment (cclauss) [reply samtranslator/model/eventsources/push.py]: My editor (Atom) removes trailing whitespace automatically on save.

Comment (sanathkr) [reply samtranslator/model/eventsources/push.py]: That's cheating! I gave credit for your work, but turns out Atom did the work 😉 lol

**Comment by cclauss** [comment]
Is there something more that needs to be done on this PR?

**Pull request merged**